sftp/sftp.go

```diff
@@ -97,10 +120,54 @@ func (fm *fileManagerImpl) Download(remoteFilePath, localDir string) error {
 
 // Delete deletes a file on the remote server
 func (fm *fileManagerImpl) Delete(remoteFilePath string) error {
+	if fm.retryOnIdleConnection {
+		return fm.retryOnConnectionLost(func() error {
+			return fm.delete(remoteFilePath)
+		})
+	}
+
+	return fm.delete(remoteFilePath)
+}
+
+func (fm *fileManagerImpl) delete(remoteFilePath string) error {
 	err := fm.client.Remove(remoteFilePath)
 	if err != nil {
 		return fmt.Errorf("cannot delete file: %w", err)
 	}
-
 	return nil
 }
+
+func (fm *fileManagerImpl) reset() error {
+	return fm.client.Reset()
+}
+
+// FileManager is not concurrent safe. It should not be used from multiple goroutines concurrently without additional synchronization.
+func NewFileManager(config *SSHConfig, opts ...Option) (FileManager, error) {
```

Comment:
Please make the comment as per good practice: https://tip.golang.org/doc/comment

If the function is called `NewFileManager` for example valid comments are:

```go
// NewFileManager function does ...
// A NewFileManager function is ...
```

You're missing the `New` part.